### PR TITLE
[Feature] Add multichoice combobox

### LINKED ISSUE #65: No deselect methods from selenium in framework's IComboBoxes
Reproduces when multiple values can be selected in a drop down. 
If you select another value, this value is added to result list, if you select the selected value (set by default), the value remains selected. So it is unable to deselect default value or deselect all values from drop down (deselectByValue/Index() and deselectAll() methods in selenium)

### PR DESCRIPTION
**Pull request by eugenenekh**
Closes #65

### CODE REVIEW

File: src/main/java/aquality/selenium/elements/ComboBox.java [hunk 1]
```diff
@@ -49,20 +49,20 @@ public void clickAndSelectByText(String value) {
...
     public void selectByContainingValue(String value) {
         logElementAction(LOG_SELECTING_VALUE, value);
-        selectOptionThatContains(element -> element.getAttribute(Attributes.VALUE.toString()),
+        applySelectFuncToOptionThatContains(element -> element.getAttribute(Attributes.VALUE.toString()),
                 Select::selectByValue,
                 value);
     }
 
-    private void selectOptionThatContains(Function<WebElement, String> getValueFunc, BiConsumer<Select, String> selectFunc, String value){
+    protected void applySelectFuncToOptionThatContains(Function<WebElement, String> getValueFunc, BiConsumer<Select, String> selectFunc, String value){
```
Comment (DmitryBogatko) [inline src/main/java/aquality/selenium/elements/ComboBox.java:65]: maybe simple 'applySelectToOptionThatContains'? without Func in the name

File: src/main/java/aquality/selenium/elements/ElementFactory.java [hunk 2]
```diff
@@ -39,6 +39,7 @@ public ElementFactory(IConditionalWait conditionalWait, IElementFinder elementFi
         typesMap.put(IButton.class, Button.class);
         typesMap.put(ICheckBox.class, CheckBox.class);
         typesMap.put(IComboBox.class, ComboBox.class);
+        typesMap.put(IMultiChoiceComboBox.class, MultiChoiceComboBox.class);
```
Comment (DmitryBogatko) [inline src/main/java/aquality/selenium/elements/ElementFactory.java:42]: maybe simple IMultiChoiceBox? Or it is necessary to emphasize that it is combobox?

File: src/test/java/w3schools/forms/SelectMultipleForm.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    private final IMultiChoiceComboBox cbxCars = getElementFactory().getMultiChoiceComboBox(By.id("cars"), "Cars");
+    private final IButton btnSubmit = getElementFactory().getButton(By.cssSelector("input[type='submit']"), "Submit");
+    private final ITextBox txbResult = getElementFactory().getTextBox(By.cssSelector(".w3-large"), "Result");
+
+    public SelectMultipleForm() {
+        super(By.id("iframe"), "Select Multiple");
+    }
+
+    public void switchToResultFrame() {
```
Comment (DmitryBogatko) [inline src/test/java/w3schools/forms/SelectMultipleForm.java:29]: I've noticed switchToResultFrame is always together with selectAll
won't be better to do this inside of selectAll?

File: src/main/java/aquality/selenium/elements/ElementFactory.java (hunk 2 above)
Comment (eugenenekh) [reply src/main/java/aquality/selenium/elements/ElementFactory.java:42]: Naming is definitely up for discussion

File: src/test/resources/TestSuite.xml [hunk 4]
```diff
@@ -7,9 +7,9 @@
             <class name="tests.integration.AlertTests"/>
             <class name="tests.integration.BrowserTests"/>
             <class name="tests.integration.BrowserTabsTests"/>
-            <class name="tests.integration.ElementTests"/>
+            <class name="tests.integration.elements.ElementTests"/>
```
Comment (mialeska) [inline src/test/resources/TestSuite.xml:10]: have you added your test class here?

File: src/main/java/aquality/selenium/elements/interfaces/IMultiChoiceComboBox.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package aquality.selenium.elements.interfaces;
+
+public interface IMultiChoiceComboBox extends IComboBox {
+
+    /**
+     * Deselect all selected options
+     */
+    void deselectAll();
```
Comment (mialeska) [inline src/main/java/aquality/selenium/elements/interfaces/IMultiChoiceComboBox.java:8]: could you add some more functions to this interface please?
As I can see, at least the following could be added:
- [ ] `void selectAll()`
- [ ] `void selectAllByContainingText(String text)`
- [ ] `void selectAllByContainingValue(String value)`
- [ ] `List<String> getSelectedTexts()`
- [ ] `List<String> getSelectedValues()`

File: src/main/java/aquality/selenium/elements/ElementFactory.java (hunk 2 above)
Comment (mialeska) [reply src/main/java/aquality/selenium/elements/ElementFactory.java:42]: +1 vote for IMultiChoiseBox

**Review by mialeska — CHANGES_REQUESTED** [review]

**Review by mialeska — APPROVED** [review]

**Pull request merged**